### PR TITLE
update for multi-storage drivers

### PR DESCRIPTION
**Pull request by jreadey**
This PR enables HSDS to support different storage back ends (posix, s3, azure blob) in the same running instance.
If the bucket query parameter is pre-appended with a storage specifier, the corresponding driver will be used for reads and writes.
Currently supported prefixes are:

- "s3://" - aws s3
- "file://" - posix 
- "https://account_name>.blob.windows.net/" - azure  blob storage

If no prefix is used, the default driver will be used.  Where default is the first one meeting the given precondition from this list:

- "file://" - if root_dir config is set
- "s3://" - if aws_s3_gateway is set
- "https....net/" - if azure_connection_string is set
- "file://" - if none of the above apply

### CODE REVIEW

File: hsds/util/domainUtil.py [hunk 1]
```diff
@@ -51,22 +51,103 @@ def isIPAddress(s):
...
+        return True
+
+    bucket = _stripProtocol(bucket)
+
+    # Bucket names must contain at least 1 character
+    if len(bucket) < 1:
+        is_valid = False
+
+    # Bucket names can consist only of alphanumeric characters, underscores, dots, and hyphens
+    # other than
```
Comment (mattjala) [inline hsds/util/domainUtil.py:92]: This comment looks to be a leftover snippet

File: hsds/datanode_lib.py [hunk 2]
```diff
@@ -327,14 +327,17 @@ async def get_metadata_obj(app, obj_id, bucket=None):
     log.info(f"get_metadata_obj: {obj_id} bucket: {bucket}")
     if isValidDomain(obj_id):
         domain_bucket = getBucketForDomain(obj_id)
+        log.debug(f"tbd - domain_bucket: {domain_bucket}")
```
Comment (mattjala) [inline hsds/datanode_lib.py:330]: Dev comment should be removed

**Review by mattjala — DISMISSED** [review]
Besides nitpicks, LGTM

File: hsds/datanode_lib.py (hunk 2 above)
Comment (jreadey) [reply hsds/datanode_lib.py:330]: done

File: hsds/util/idUtil.py [hunk 3]
```diff
@@ -21,6 +21,43 @@
 from .. import hsds_logger as log
 
 
+S3_URI = "s3://"
+FILE_URI = "file://"
+AZURE_URI = "blob.core.windows.net/"  # preceded with "https://"
+
+
+def _getStorageProtocol(uri):
+    """ returns 's3://', 'file://', or 'https://...net/' prefix if pressent.
```
[the remark this answers is not in the document]
Comment (jreadey) [reply hsds/util/idUtil.py]: fixed

**Pull request merged**